### PR TITLE
Support for optional SYNAPSE_DB_PORT env var

### PR DESCRIPTION
**Pull request by richvdh**
Add an env var to permit connecting to postgres on a different port

### CODE REVIEW

File: retention/cohort_analysis.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def get_conn(self):
         conn = connect(
             dbname=self.SYNAPSE_DB_DATABASE,
             user=self.SYNAPSE_DB_USERNAME,
+            port=os.environ.get("SYNAPSE_DB_PORT"),
```
Comment (callahad) [inline retention/cohort_analysis.py]: Shouldn't this follow the same pattern as the other variables, which are set in the constructor immediately above?

**Review by callahad — CHANGES_REQUESTED** [review]

File: retention/cohort_analysis.py (hunk 1 above)
Comment (richvdh) [reply retention/cohort_analysis.py]: well maybe, but that seems entirely redundant. I could refactor the others if you want?

Comment (callahad) [reply retention/cohort_analysis.py]: Either way, as long as it's consistent :)

Comment (richvdh) [reply retention/cohort_analysis.py]: fair enough.

I ended up refactoring it to build a DSN in the constructor (which is what `connect` does internally if you pass it a load of kwargs). WDYT?

**Review by callahad — APPROVED** [review]
Works for me

**Pull request merged**